pilot/pkg/proxy/resolve_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"testing"
+)
+
+func TestResolveAddr(t *testing.T) {
+	var testCases = []struct {
+		name     string
+		input    string
+		expected string
+		errStr   string
+	}{
+		{
+			name:     "Host by name",
+			input:    "localhost:9080",
+			expected: "127.0.0.1:9080",
+			errStr:   "",
+		},
+		{
+			name:     "Host by name w/brackets",
+			input:    "[localhost]:9080",
+			expected: "127.0.0.1:9080",
+			errStr:   "",
+		},
+		{
+			name:     "Host by IPv4",
+			input:    "127.0.0.1:9080",
+			expected: "127.0.0.1:9080",
+			errStr:   "",
+		},
+		{
+			name:     "Host by IPv6",
+			input:    "[::1]:9080",
+			expected: "[::1]:9080",
+			errStr:   "",
+		},
+		{
+			name:     "Bad IPv4",
+			input:    "127.0.0.1.1:9080",
+			expected: "",
+			errStr:   "lookup failed for IP address: lookup 127.0.0.1.1: no such host",
+		},
+		{
+			name:     "Bad IPv6",
+			input:    "[2001:db8::bad::1]:9080",
+			expected: "",
+			errStr:   "lookup failed for IP address: lookup 2001:db8::bad::1: no such host",
+		},
+		{
+			name:     "Empty host",
+			input:    "",
+			expected: "",
+			errStr:   "",
```

Comment:
This should be an error, right?

Reply:
In the existing code, they do not consider this an error. The comments in the calling logic in pilot-agent/main.go seems to indicate that this would be ignored:

`
      // If istio-mixer.istio-system can't be resolved, skip generating the statsd config.
      // (instead of crashing). Mixer is optional.
`

So, I left the behavior the same.

Reply:
(nit) I would change the call site ?

Reply:
Do you mean, return an error here and have the caller check and ignore the error? In that case , would the recommended way be to check the error message for a specific clause like "no address specified", and the ignore the error?